File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        if(k == s.length())return s.size();
        int ans=0,maxf=0,i=0,l=0;
        int hash[26] = { 0 };
        while(s[i]!= '\0'){
             hash[s[i]-'A']++;
             maxf = max(maxf,hash[s[i]-'A']);

             while((i-l+1 - maxf) > k){
                hash[s[l]-'A']--;l++;
             }

             if((i-l+1 - maxf) <= k){
              ans = max(ans,i-l+1);
             }
             i++;
        }
        return ans;
    }
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
   
    public int characterReplacement(String s, int k) {
       
       int r = 0, l = 0, ans = 0, maxf = 0;
       int[] hash = new int[26];
       while(r<s.length()){
          hash[s.charAt(r) - 'A']++;
          maxf = Math.max(maxf,hash[s.charAt(r) - 'A']);

          while(((r-l+1) - maxf) > k){
            hash[s.charAt(l) - 'A']--;
            maxf = 0;
            for(int i=0;i<=25;i++){
               maxf = Math.max(maxf,hash[i]);
            }
            l++;
          }
         if(((r-l+1) - maxf) <= k){
            ans = Math.max(ans,r-l+1 );
         }
         r++;


       }

       return ans;
    }
}